pkg/standalone-utils/test/VaultActionsStableJoinExit.test.ts

```diff
@@ -0,0 +1,757 @@
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/dist/src/signer-with-address';
+import Vault from '@balancer-labs/v2-helpers/src/models/vault/Vault';
+import { fp } from '@balancer-labs/v2-helpers/src/numbers';
+import TokenList from '@balancer-labs/v2-helpers/src/models/tokens/TokenList';
+import StablePool from '@balancer-labs/v2-helpers/src/models/pools/stable/StablePool';
+import { getPoolAddress, StablePoolEncoder } from '@balancer-labs/balancer-js';
+import { MAX_UINT256, ZERO_ADDRESS } from '@balancer-labs/v2-helpers/src/constants';
+import { expectBalanceChange } from '@balancer-labs/v2-helpers/src/test/tokenBalance';
+import * as expectEvent from '@balancer-labs/v2-helpers/src/test/expectEvent';
+import { expectTransferEvent } from '@balancer-labs/v2-helpers/src/test/expectTransfer';
+import { Contract } from 'ethers';
+import { expect } from 'chai';
+import Token from '@balancer-labs/v2-helpers/src/models/tokens/Token';
+import { Zero } from '@ethersproject/constants';
+import {
+  expectChainedReferenceContents,
+  setChainedReferenceContents,
+  toChainedReference,
+} from './helpers/chainedReferences';
+import TypesConverter from '@balancer-labs/v2-helpers/src/models/types/TypesConverter';
+import { Account } from '@balancer-labs/v2-helpers/src/models/types/types';
+import {
+  setupRelayerEnvironment,
+  encodeJoinPool,
+  encodeExitPool,
+  PoolKind,
+  approveVaultForRelayer,
+  getJoinExitAmounts,
+  encodeSwap,
+} from './VaultActionsRelayer.setup';
+
+describe('stable pools', () => {
+  let vault: Vault;
+  let tokens: TokenList;
+  let relayer: Contract, relayerLibrary: Contract;
+  let user: SignerWithAddress, other: SignerWithAddress;
+  let sender: Account, recipient: Account;
+  let poolIdStable: string;
+  let poolIdStable2: string;
+  let bptIndex: number;
+  let bptIndex2: number;
+
+  before('setup environment', async () => {
+    ({ user, other, vault, relayer, relayerLibrary } = await setupRelayerEnvironment());
+  });
+
+  before('setup common recipient', () => {
+    // All the tests use the same recipient; this is a simple abstraction to improve readability.
+    recipient = user;
+  });
```

Comment:
Make this a separate account, e.g. using `randomAddress()` if it doesn't send any transactions - it can be confusing and misleading if we're using the same account under two names.

Reply:
In the old VaultActions too?

Reply:
Yes!